packages/wdio-config/src/dataproviders.js

```diff
@@ -0,0 +1,41 @@
+import { safeRequire } from '@wdio/utils'
+import ConfigParser from './lib/ConfigParser'
+
+/**
+ * Loads the data providers from the files to memory
+ * @param  {Object} dataProviderFilesPaths  List of files with data provider injections
+ */
+export function initializeDataProviders (dataProviderFilesPaths) {
+    let dataProviderFiles =  ConfigParser.getFilePaths(dataProviderFilesPaths)
+    let dataProvidersMap = {}
+    let currentDataProviderFile = ''
+
+    global.dataProvider = function (specFilePath, dataSet) {
+        if (!Array.isArray(dataSet)) {
+            throw new Error(`The data set passed from the file "${currentDataProviderFile}" for the spec file path "${specFilePath}" is not an array.`)
+        }
+
+        let specFiles =  ConfigParser.getFilePaths(specFilePath)
+        specFiles.forEach((specFile) => {
+            let specFileDataProvider = dataProvidersMap[specFile]
+            if(typeof specFileDataProvider == 'undefined') {
+                let dataProviderObject = {}
+                dataProviderObject.specFilePath = specFilePath
+                dataProviderObject.dataSet = dataSet
+                dataProviderObject.dataProviderFile = currentDataProviderFile
+                dataProvidersMap[specFile] = dataProviderObject
+            } else {
+                throw new Error(`With spec file path "${specFilePath}" in the data provider "${currentDataProviderFile}", you are attempting to override the data set already defined for the spec "${specFile}" from data provider "${specFileDataProvider.dataProviderFile}" using spec file path "${specFileDataProvider.specFilePath}. Please resolve conflict between these dataProvider functions.`)
+            }
+        })
+    }
+
+    for (let dpFile of dataProviderFiles) {
+        currentDataProviderFile = dpFile
+        safeRequire(dpFile)
+    }
+
+    delete global.dataProvider
```

Comment:
Why do we register this globally in the first place?

Reply:
any answer to this?

Reply:
There are only 2 possible ways to inject the test data in to the WDIO runner
1. Through config file
2. Through a global method

We preferred the 2nd approach to inject the data to the runner, because the data injection logic could be complex. Once the data is injected, I didn't want any spec to inject any more data using this global method during the test execution. That's why, I am deleting the global method.

Since the

Reply:
@christian-bromann - how can we create a blog post here - https://webdriver.io/blog/